config/routes.rb

```diff
@@ -1,11 +1,13 @@
 Rails.application.routes.draw do
 
+
   devise_for :users
   resources :users, only: [:update]
    
-   resources :topics do
-     resources :posts, except: [:index]
-   end
+   resources :topics, :posts, :comments do
+   resources :posts, except: [:index]
+   resources :comments, only: [:create]
```

Comment:
No. posts is nested under topics, this is what you have:

```
resources :topics do
  resources :posts, except: [:index]
 end
```

Just nest comments under posts:

```
resources :topics do
  resources :posts, except: [:index] do
    resources :comments, only: [:create]
  end
 end
```

Don't forget to run `rake routes` and adjust accordingly.